internal/gapicgen/cmd/genbot/alias.go

```diff
@@ -0,0 +1,176 @@
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//go:build !windows
+// +build !windows
+
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"cloud.google.com/go/internal/aliasfix"
+	"cloud.google.com/go/internal/aliasgen"
+	"cloud.google.com/go/internal/gapicgen/execv/gocmd"
+	"cloud.google.com/go/internal/gapicgen/git"
+	"golang.org/x/sync/errgroup"
+)
+
+type aliasConfig struct {
+	githubAccessToken string
+	githubUsername    string
+	githubName        string
+	githubEmail       string
+	gocloudDir        string
+	genprotoDir       string
+}
+
+func genAliasMode(ctx context.Context, c aliasConfig) error {
+	for k, v := range map[string]string{
+		"githubAccessToken": c.githubAccessToken,
+		"githubUsername":    c.githubUsername,
+		"githubName":        c.githubName,
+		"githubEmail":       c.githubEmail,
+	} {
+		if v == "" {
+			log.Printf("missing or empty value for required flag --%s\n", k)
+			flag.PrintDefaults()
+		}
+	}
+
+	// Setup the client and git environment.
+	githubClient, err := git.NewGithubClient(ctx, c.githubUsername, c.githubName, c.githubEmail, c.githubAccessToken)
+	if err != nil {
+		return err
+	}
+
+	// Check current alias regen status.
+	if pr, err := githubClient.GetPRWithTitle(ctx, "go-genproto", "", "auto-regenerate alias files"); err != nil {
+		return err
+	} else if pr != nil && pr.IsOpen {
+		log.Println("there is already an alias re-generation in progress")
+		return nil
+	} else if pr != nil && !pr.IsOpen {
+		lastWeek := time.Now().Add(-7 * 24 * time.Hour)
+		if pr.Created.After(lastWeek) {
+			log.Println("there is already an alias re-generation already this week")
+			return nil
+		}
+	}
+
+	// Clone repositories if needed.
+	log.Println("creating temp dir")
+	tmpDir, err := ioutil.TempDir("", "update-genproto")
+	if err != nil {
+		log.Fatal(err)
+	}
+	log.Printf("temp dir created at %s\n", tmpDir)
+	grp, _ := errgroup.WithContext(ctx)
+	grp.Go(func() error {
+		if c.genprotoDir != "" {
+			return nil
+		}
+		return git.DeepClone("https://github.com/googleapis/go-genproto", filepath.Join(tmpDir, "genproto"))
+	})
+	grp.Go(func() error {
+		if c.gocloudDir != "" {
+			return nil
+		}
+		return git.DeepClone("https://github.com/googleapis/google-cloud-go", filepath.Join(tmpDir, "gocloud"))
+	})
+	if err := grp.Wait(); err != nil {
+		return err
+	}
+	genprotoDir := deafultDir(filepath.Join(tmpDir, "genproto"), c.genprotoDir)
```

Comment:
Typo: `deafultDir`. Sorry I missed this when the function was initially reviewed.

Reply:
done.